components/validator/guardian.go

```diff
@@ -0,0 +1,159 @@
+	securityCouncilContract, err := bindings.NewSecurityCouncil(cfg.SecurityCouncilAddr, cfg.L1Client)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(parentCtx)
+
+	return &Guardian{
+		log:                     l,
+		cfg:                     cfg,
+		ctx:                     ctx,
+		cancel:                  cancel,
+		callOpts:                utils.NewCallOptsWithSender(ctx, cfg.TxManager.From()),
+		txOpts:                  utils.NewSimpleTxOpts(ctx, cfg.TxManager.From(), cfg.TxManager.Signer),
+		securityCouncilContract: securityCouncilContract,
+	}, nil
+}
+
+func (g *Guardian) Stop() {
+	g.log.Info("stop Guardian")
+
+	if g.securityCouncilSub != nil {
+		g.securityCouncilSub.Unsubscribe()
+	}
+}
+
+func (g *Guardian) handleValidationRequested(logs chan *bindings.SecurityCouncilValidationRequested) {
+	defer func() {
+		g.log.Info("exit handleRequestValidated goroutine")
+		close(logs)
+		g.wg.Done()
+	}()
+	for {
+		select {
+		case ev := <-logs:
+			if !g.isSecurityCouncil(g.callOpts.From) {
+				continue
+			}
+			g.wg.Add(1)
+			go g.processOutputValidation(ev)
+		case <-g.ctx.Done():
+			return
+		}
+	}
+}
+
+func (g *Guardian) processOutputValidation(event *bindings.SecurityCouncilValidationRequested) {
+	g.log.Info("start processOutputValidation")
+	defer func() {
+		g.log.Info("exit processOutputValidation goroutine")
+		g.wg.Done()
+	}()
+
+	isValidate, err := g.isL2OutputValidate(event.OutputRoot, event.L2BlockNumber.Uint64())
+	if err != nil {
+		g.log.Error("isL2OutputValidate failed", "err", err)
+		return
+	}
+
+	if isValidate {
+		g.log.Info("confirmTransaction")
+		_, err := g.securityCouncilContract.ConfirmTransaction(g.txOpts, event.TransactionId)
+		if err != nil {
+			g.log.Error("tx call ConfirmTransaction failed", "err", err)
+			return
+		}
+	}
+}
+
+func (g *Guardian) Start() {
+	g.log.Info("start Guardian")
+
+	watchOpts := &bind.WatchOpts{Context: g.ctx, Start: nil}
+	logRespChan := make(chan *bindings.SecurityCouncilValidationRequested)
+	g.securityCouncilSub = event.ResubscribeErr(time.Second*10, func(ctx context.Context, err error) (event.Subscription, error) {
+		if err != nil {
+			g.log.Error("resubscribing after failed SecurityCouncilValidationRequested event", "err", err)
+		}
+		return g.securityCouncilContract.WatchValidationRequested(watchOpts, logRespChan, nil)
+	})
+	g.wg.Add(1)
+	go g.handleValidationRequested(logRespChan)
+}
+
+func (g *Guardian) isSecurityCouncil(address common.Address) bool {
+	g.log.Info("isSecurityCouncil")
+	ownerAddresses, err := g.securityCouncilContract.GetOwners(g.callOpts)
+
+	if err != nil {
+		g.log.Error("isSecurityCouncil call failed", "err", err)
+	}
+
+	isOwner := func(addressList []common.Address, addr common.Address) bool {
+		for _, item := range addressList {
+			if item == addr {
+				return true
+			}
+		}
+		return false
+	}(ownerAddresses, address)
```

Comment:
Please extract this util functions to somewhere to be shared across the codes using generic.